lib/include/domain_participant/builtin/discovery_config/DiscoveryConfig.hpp

```diff
@@ -0,0 +1,1057 @@
+ */
+FASTDDS_QOS_PROFILES_MANAGER_DllAPI void set_ignore_participant_flags(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& ignore_participant_flags);
+
+/**
+ * @brief Set the Domain Participant builtin discovery EDP flag.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] profile_id Domain participant profile identifier.
+ * @param[in] edp Builtin discovery EDP flag.
+ *
+ * @throw ElementInvalid Exception if the provided flag is not valid.
+ */
+FASTDDS_QOS_PROFILES_MANAGER_DllAPI void set_edp(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& edp);
+
+/**
+ * @brief Set the Domain Participant builtin discovery EDP flag.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] profile_id Domain participant profile identifier.
+ * @param[in] edp Builtin discovery EDP flag.
+ *
+ * @throw ElementInvalid Exception if the provided flag is not valid.
+ */
+FASTDDS_QOS_PROFILES_MANAGER_DllAPI void set_edp(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& edp);
+
+/**
+ * @brief Set the Domain Participant builtin discovery simple EDP configuration flag.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] profile_id Domain participant profile identifier.
+ * @param[in] simple_edp_pubwriter_subreader Builtin discovery simple EDP configuration flag.
+ *
+ * @throw ElementInvalid Exception if the provided flag is not valid.
+ */
+FASTDDS_QOS_PROFILES_MANAGER_DllAPI void set_simple_edp_pubwriter_subreader(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& simple_edp_pubwriter_subreader);
+
+/**
+ * @brief Set the Domain Participant builtin discovery simple EDP configuration flag.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] profile_id Domain participant profile identifier.
+ * @param[in] simple_edp_pubreader_subwriter Builtin discovery simple EDP configuration flag.
+ *
+ * @throw ElementInvalid Exception if the provided flag is not valid.
+ */
+FASTDDS_QOS_PROFILES_MANAGER_DllAPI void set_simple_edp_pubreader_subwriter(
+        const std::string& xml_file,
+        const std::string& profile_id,
+        const std::string& simple_edp_pubreader_subwriter);
+
+/**
+ * @brief Set the Domain Participant builtin discovery lease duration.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] profile_id Domain participant profile identifier.
+ * @param[in] duration_sec Builtin discovery lease duration seconds.
+ * @param[in] duration_nanosec Builtin discovery lease duration nanoseconds.
+ *
+ * @throw ElementInvalid Exception if the provided lease duration arguments are not valid.
+ */
+FASTDDS_QOS_PROFILES_MANAGER_DllAPI void set_lease_duration(
```

Comment:
This `set` function with both `sec` and `nanosec` arguments is no longer required, we changed it into two individual set functions for `sec` and `nanosec`